lib/cocoapods/sources_manager.rb

```diff
@@ -70,15 +71,27 @@ def create_source_with_url(url)
       #         The URL of the source.
       #
       def cdn_url?(url)
-        if url =~ %r{^https?:\/\/}
-          require 'typhoeus'
-
-          response = Typhoeus.get(url + '/CocoaPods-version.yml', :netrc_file => Netrc.default_path, :netrc => :optional)
-          response.code == 200 && begin
-            response_hash = YAML.load(response.body) # rubocop:disable Security/YAMLLoad
-            response_hash.is_a?(Hash) && !Source::Metadata.new(response_hash).latest_cocoapods_version.nil?
-          end
-        end
+        return unless url =~ %r{^https?:\/\/}
+
+        uri_options = {
+          :redirect => false,
+        }
+
+        netrc_info = Netrc.read
+        netrc_host = URI.parse(url).host
+        credentials = netrc_info[netrc_host]
+        uri_options[:http_basic_authentication] = credentials if credentials
+
+        response = OpenURI.open_uri(URI.join(url, 'CocoaPods-version.yml'), uri_options)
+
+        response_hash = YAML.load(response.read) # rubocop:disable Security/YAMLLoad
+        response_hash.is_a?(Hash) && !Source::Metadata.new(response_hash).latest_cocoapods_version.nil?
+      rescue ::OpenURI::HTTPError
```

Comment:
can rescue the three error classes in the same rescue statement

Reply:
will do